src/Compilers/CSharp/csc/Csc.cs

```diff
@@ -46,6 +47,11 @@ internal static int Run(string[] args)
             }
         }
 
+        public override Assembly LoadAssembly(string fullPath)
+        {
+            return Assembly.Load(fullPath);
+        }
```

Comment:
You want `Assembly.LoadFrom(string assemblyFile)`.

Right now this calls `Assembly.Load(string assemblyName)` where `assemblyName` is the full assembly name, e.g., `"SampleAssembly, Version=1.0.2004.0, Culture=neutral, PublicKeyToken=8744b20f8da049e3"`. Passing in a path to an assembly will end in tears.

Also note that a straight-up `Assembly.LoadFrom(string assemblyFile)` will be fine for csc.exe and vbc.exe as those are short-lived processes and it is OK for the assembly to be locked while they run, but vbcscompiler.exe needs to avoid locking them. Shadow-copying will be required.